src/test/java/com/mabl/integration/jenkins/MablRestApiClientTest.java

```diff
@@ -118,6 +118,25 @@ public void createDeploymentOnlyLabelsHappyPathTest() throws IOException {
         assertSuccessfulCreateDeploymentRequest(fakeRestApiKeyId, environmentId, applicationId, labels);
     }
 
```

Comment:
Perhaps this logic can be tested w/ an API mock?

It could return three different results. The first case saying that the execution is complete, the second saying that it's really doing a retry, and the third saying the retry is complete?

Reply:
Great idea. Will add it, thank you!